### PR TITLE
[IMP][10.0] medical: Simplify models

### PR DESCRIPTION
**Pull request by lasley**
Yeah sorry for all the code in this one. I made some changes that cascaded to other modules, then ran into some issues that were fixed on OCA/9.0 already so had to forward-port. Then I ran into other things that needed to be grep'd, and yeah so uhhh whoops.

[IMP] medical: Simplify models  …
* Add `partner_contact_birthdate`, `partner_firstname`, and `partner_identification`, `partner_contact_gender` dependencies
* Remove attributes in partner and patient that are duplicated by dependencies
* Remove is_* attributes on partner in favor of `type`
* Act upon type to create medical entities automatically from partner if of correct type
* Change `dob` to `birthdate_date` for compatibility with `partner_contact_birthdate`
* Change `dod` to `date_death`
* Change `deceased` to `is_deceased`
* Forward port OCA 9.0
* Update patient entity type

[IMP] medical_base_us: Remove constraints
* Remove constraints in favor of partner_identification validation
* Add `res.partner.category_id` validators for dea, npi, and ssn
* Remove security for test models that no longer exist

[IMP] medical_pharmacy:
* Forward port OCA/9.0
* Update entity type

[IMP] medical_physician:
* Remove medical_constants
* Update Entity type
* Forward port OCA/9.0

[IMP] medical_medicament: Forward port OCA/9.0
[IMP] medical_pathology: Forward port OCA/9.0
[IMP] medical_patient_disease: Optimistic forward port of pending OCA
[IMP] medical_medication: Optimistic forward port of pending OCA
[IMP] medical_prescription: Optimistic forward port of pending OCA
[IMP] medical_manufacturer: Update entity type
[IMP] medical_insurance:  Update entity type
[IMP] medical_*: Update naming to correspond with patients renames

[REM] medical_patient_us: Disable module
[REM] medical_physician_us: Disable module
[REM] medical_pharmacy_us: Disable module

This is dependent on:
- [x] OCA/partner-contact#315

### CODE REVIEW

**Comment by lasley** [comment]
OMFG it works. Ready for review!

Note: I'm like 99% positive the coverage drop is due to there being less code.

File: medical_prescription/tests/test_medical_prescription_order.py [hunk 1]
```diff
@@ -9,24 +9,27 @@ class TestMedicalPrescriptionOrder(TransactionCase):
...
-    def test_active_false(self):
-        """ It should set active to true when rx line is active """
-        order = self.line.prescription_order_id
-        for line in order.prescription_order_line_ids:
-            line.active = False
+    def test_compute_active_no_line_ids(self):
+        """ Test active is False if no rx line_ids """
         self.assertFalse(
-            self.line.prescription_order_id.active
+            self.rx_6.active,
```
Comment (tedsalmon) [inline medical_prescription/tests/test_medical_prescription_order.py]: `medical_prescription_prescription_order_6` has no `prescription_order_line_ids` which according to medical_prescription/models/medical_prescription_order.py:_compute_active results in the RX being `active=True`

**Review by tedsalmon — CHANGES_REQUESTED** [review]
There's a test that needs changing and is causing the build to fail. I also committed a change to fix a broken demo data reference. 😄 

EDIT: Additionally, there are a large number of .py files that are having their mode altered to `100755` 😢

**Comment by lasley** [comment]
Test & perms fixed

**Pull request merged**